lib/cache.h

```diff
@@ -36,10 +36,13 @@
 #ifndef __METAL_CACHE__H__
 #define __METAL_CACHE__H__
 
+#include <metal/system/@PROJECT_SYSTEM@/cache.h>
+
```

Comment:
This change will impact other OS implementation, the other OS needs cache.h too

Reply:
I will make the changes, but wanted to see if the general approach was ok first.

Reply:
For example should we use metal_machine_cache_flush or __metal_cache_flush

Reply:
This sounds a good approach and we can use this approach to improve some other APIs too.